0b.md

```diff
@@ -0,0 +1,273 @@
+  ```
+
+- Or otherwise, when master account decides to not allow on-behalf for `kind 7`, but keep `kind 1` active:
+
+  ```json
+  {
+    "id": "567b41fc9060c758c4216fe5f8d3df7c57daad7ae757fa4606f0c39d4dd220ef",
+    "kind": "10100",
+    "pubkey": "8e0d3d3eb2881ec137a11debe736a9086715a8c8beeeda615780064d68bc25dd",
+    "tags": [
+      ["p",
+      "477318cfb5427b9cfc66a9fa376150c1ddbc62115ae27cef72417eb959691396",
+      "",
+      "active:1674834236:1,7"],
+      ["p",
+      "477318cfb5427b9cfc66a9fa376150c1ddbc62115ae27cef72417eb959691396",
+      "",
+      "active:1721934607:1"]
+    ],
+    "content": "",
+    ...
+    "sig": "a9a4e2192eede77e6c9d24ddfab95ba3ff7c03fbd07ad011fff245abea431fb4d3787c2d04aad001cb039cb8de91d83ce30e9a94f82ac3c5a2372aa1294a96bd"
+  }
+  ```
+
+- In the case that master decides to not permit on-behalf publishing for that active account anymore, but keep the already published events, for instance if the device that was using that active account is being substituted, and that key safely destroyed:
+
+  ```json
+  {
+    "id": "567b41fc9060c758c4216fe5f8d3df7c57daad7ae757fa4606f0c39d4dd220ef",
+    "kind": "10100",
+    "pubkey": "8e0d3d3eb2881ec137a11debe736a9086715a8c8beeeda615780064d68bc25dd",
+    "tags": [
+      ["p",
+      "477318cfb5427b9cfc66a9fa376150c1ddbc62115ae27cef72417eb959691396",
+      "",
+      "active:1674834236:1,7"],
+      ["p",
+      "477318cfb5427b9cfc66a9fa376150c1ddbc62115ae27cef72417eb959691396",
+      "",
+      "active:1721934607:1"],
+      ["p",
+      "477318cfb5427b9cfc66a9fa376150c1ddbc62115ae27cef72417eb959691396",
+      "",
+      "inactive:1722343578"]
+    ],
+    "content": "",
+    ...
+    "sig": "a9a4e2192eede77e6c9d24ddfab95ba3ff7c03fbd07ad011fff245abea431fb4d3787c2d04aad001cb039cb8de91d83ce30e9a94f82ac3c5a2372aa1294a96bd"
+  }
+  ```
+
+- Or instead, if the active account got compromised, and it is best to consider all content published from that account,on-behalf of master account, to be compromised as well:
+
+  ```json
+  {
+    "id": "567b41fc9060c758c4216fe5f8d3df7c57daad7ae757fa4606f0c39d4dd220ef",
+    "kind": "10100",
+    "pubkey": "8e0d3d3eb2881ec137a11debe736a9086715a8c8beeeda615780064d68bc25dd",
+    "tags": [
+      ["p",
+      "477318cfb5427b9cfc66a9fa376150c1ddbc62115ae27cef72417eb959691396",
+      "",
+      "active:1674834236:1,7"],
+      ["p",
+      "477318cfb5427b9cfc66a9fa376150c1ddbc62115ae27cef72417eb959691396",
+      "",
+      "active:1721934607:1"],
+      ["p",
+      "477318cfb5427b9cfc66a9fa376150c1ddbc62115ae27cef72417eb959691396",
+      "",
+      "revoked:1722343578"]
+    ],
+    "content": "",
+    ...
+    "sig": "a9a4e2192eede77e6c9d24ddfab95ba3ff7c03fbd07ad011fff245abea431fb4d3787c2d04aad001cb039cb8de91d83ce30e9a94f82ac3c5a2372aa1294a96bd"
+  }
+  ```
+
+
+### Relay & Client Support
+
+Relays should answer requests such as `["REQ", "", {"authors": ["A"]}]` by querying both the `pubkey` and on-behalf `#b` tags `[1]` value.
+
```

Comment:
NIP-42 MUST allow the use of the subkey to login if authorized.

Reply:
This is cool. So you mean that kind `22242` ephemeral event on the `AUTH` message should use the `b` tag to reference the master account, and the relays should authenticate the main account when such `b` tag is present?